actions/get-mcms-tools/action.yaml

```diff
@@ -0,0 +1,100 @@
+name: Get MCMS Tools (Prebuilt)
+description:
+  Downloads and extracts the prebuilt mcms-tools CLI binary from the latest (or
+  specified) GitHub release, exposing the binary path as an output.
+
+inputs:
+  github_token:
+    description: GitHub token with access to the mcms-tools repository.
+    required: true
+  version:
+    description: "Release version (e.g., v0.3.0). required."
+    required: false
+    default: ""
+
+outputs:
+  binary_path:
+    description: "Path to the mcms binary"
+    value: ${{ steps.set_path.outputs.binary_path }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Set up platform and asset name
+      id: platform
+      shell: bash
+      run: |
+        OS=$(uname -s | tr '[:upper:]' '[:lower:]')
+        ARCH=$(uname -m)
+        if [ "$OS" = "darwin" ]; then
+          if [ "$ARCH" = "x86_64" ]; then
+            ASSET="darwin_amd64_darwin_amd64_v1.tar.gz"
+          elif [ "$ARCH" = "arm64" ] || [ "$ARCH" = "aarch64" ]; then
+            ASSET="darwin_arm64_darwin_arm64_v8.0.tar.gz"
+          else
+            echo "::error::Unsupported macOS architecture: $ARCH"
+            exit 1
+          fi
+        elif [ "$OS" = "linux" ]; then
+          if [ "$ARCH" = "x86_64" ]; then
+            ASSET="linux_amd64_linux_amd64_v1.tar.gz"
+          else
+            echo "::error::Unsupported Linux architecture: $ARCH"
+            exit 1
+          fi
+        else
+          echo "::error::Unsupported OS: $OS"
+          exit 1
+        fi
+        echo "asset_name=$ASSET" >> $GITHUB_OUTPUT
+
+    - name: Download mcms-tools Release Asset
+      id: download
+      shell: bash
+      env:
+        GH_TOKEN: ${{ inputs.github_token }}
+        INPUT_VERSION: ${{ inputs.version }}
+      run: |
+        set -e
+        REPO="smartcontractkit/mcms-tools"
+        VERSION="$INPUT_VERSION"
+        export ASSET_NAME="${{ steps.platform.outputs.asset_name }}"
```

Comment:
Some of these these variables could be set in the `env` block above.

```
REPO: "smartcontractkit/mcms-tools"
ASSET_NAME: ${{ steps.platform.outputs.asset_name }}
```

In fact, you should generally avoid using any `${{ ... }}` syntax inside shell blocks, because they can be used to escape the shell and inject anything.